src/Language/DifferentialDatalog/Syntax.hs

```diff
@@ -659,25 +590,53 @@ funcShowProto Function{..} = render $
 funcTypeVars :: Function -> [String]
 funcTypeVars = nub . concatMap (typeTypeVars . fieldType) . funcArgs
 
-data DatalogProgram = DatalogProgram { progTypedefs  :: M.Map String TypeDef
+data ModuleName = ModuleName {modulePath :: [String]}
+                  deriving (Eq, Ord)
+
+instance PP ModuleName where
+    pp (ModuleName p) = hcat $ punctuate "." $ map pp p
+
+instance Show ModuleName where
+    show = render . pp
+
+-- | Import statement
+data Import = Import { importPos    :: Pos
+                     , importModule :: ModuleName
+                     , importAlias  :: ModuleName 
+                     }
+
+instance WithPos Import where
+    pos = importPos
+    atPos i p = i { importPos = p }
+
+instance PP Import where
+    pp Import{..} = "import" <+> pp importModule <+> "as" <+> pp importAlias
```

Comment:
does this work when there is no alias?

Reply:
good catch! Imports are always eliminated before printing, which is why I did not spot this.